lib/Twig/Extension/Escaper.php

```diff
@@ -87,3 +201,277 @@ function twig_raw_filter($string)
 {
     return $string;
 }
+
+/**
+ * Escapes a string.
+ *
+ * @param Twig_Environment $env
+ * @param mixed            $string     The value to be escaped
+ * @param string           $strategy   The escaping strategy
+ * @param string           $charset    The charset
+ * @param bool             $autoescape Whether the function is called by the auto-escaping feature (true) or by the developer (false)
+ *
+ * @return string
+ */
+function twig_escape_filter(Twig_Environment $env, $string, $strategy = 'html', $charset = null, $autoescape = false)
+{
+    if ($autoescape && $string instanceof Twig_Markup) {
+        return $string;
+    }
+
+    if (!is_string($string)) {
+        if (is_object($string) && method_exists($string, '__toString')) {
+            $string = (string) $string;
+        } elseif (in_array($strategy, array('html', 'js', 'css', 'html_attr', 'url'))) {
+            return $string;
+        }
+    }
+
+    if (null === $charset) {
+        $charset = $env->getCharset();
+    }
+
+    $escapers = $env->getExtension('Twig_Extension_Escaper')->getEscapers();
+
+    if (isset($escapers[$strategy])) {
+        return $escapers[$strategy]($env, $string, $charset);
+    }
+
+    $validStrategies = implode(', ', array_keys($escapers));
+
+    throw new Twig_Error_Runtime(sprintf('Invalid escaping strategy "%s" (valid ones: %s).', $strategy, $validStrategies));
+}
+
+/**
+ * @internal
+ */
+function twig_escape_html(Twig_Environment $env, $string, $charset)
+{
+    // see http://php.net/htmlspecialchars
+
+    // Using a static variable to avoid initializing the array
+    // each time the function is called. Moving the declaration on the
+    // top of the function slow downs other escaping strategies.
```

Comment:
this comment is irrelevant as it is a separate function now, and this *is* at the top